sdk/core/azure-core/src/main/java/com/azure/core/util/implementation/FileContent.java

```diff
@@ -0,0 +1,131 @@
+import com.azure.core.util.serializer.ObjectSerializer;
+import com.azure.core.util.serializer.TypeReference;
+import reactor.core.Exceptions;
+import reactor.core.publisher.Flux;
+import reactor.core.scheduler.Scheduler;
+import reactor.core.scheduler.Schedulers;
+
+import java.io.BufferedInputStream;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * A {@link BinaryDataContent} backed by a file.
+ */
+public final class FileContent extends BinaryDataContent {
+    private static final ClientLogger LOGGER = new ClientLogger(FileContent.class);
+    private final Path file;
+    private final int chunkSize;
+    private final long length;
+    private final AtomicReference<byte[]> bytes = new AtomicReference<>();
+
+    /**
+     * Creates a new instance of {@link FileContent}.
+     *
+     * @param file The {@link Path} content.
+     * @param chunkSize The requested size for each read of the path.
+     */
+    public FileContent(Path file, int chunkSize) {
+        Objects.requireNonNull(file, "'file' cannot be null.");
+
+        if (chunkSize <= 0) {
+            throw LOGGER.logExceptionAsError(new IllegalArgumentException(
+                    "'chunkSize' cannot be less than or equal to 0."));
+        }
+        this.file = file;
+        this.chunkSize = chunkSize;
+        if (!file.toFile().exists()) {
+            throw LOGGER.logExceptionAsError(new UncheckedIOException(
+                    new FileNotFoundException("File does not exist " + file)));
+        }
+
+        this.length = file.toFile().length();
+    }
+
+    @Override
+    public Long getLength() {
+        return this.length;
+    }
+
+    @Override
+    public String toString() {
+        return new String(toBytes(), StandardCharsets.UTF_8);
+    }
+
+    @Override
+    public byte[] toBytes() {
+        bytes.compareAndSet(null, getBytes());
+        byte[] data = this.bytes.get();
+        return Arrays.copyOf(data, data.length);
+    }
+
+    @Override
+    public <T> T toObject(TypeReference<T> typeReference, ObjectSerializer serializer) {
+        return serializer.deserialize(toStream(), typeReference);
+    }
+
+    @Override
+    public InputStream toStream() {
+        try {
+            return new BufferedInputStream(new FileInputStream(file.toFile()), chunkSize);
+        } catch (FileNotFoundException e) {
+            throw LOGGER.logExceptionAsError(new UncheckedIOException("File not found " + file, e));
+        }
+    }
+
+    @Override
+    public ByteBuffer toByteBuffer() {
+        return ByteBuffer.wrap(toBytes()).asReadOnlyBuffer();
```

Comment:
I wonder if there is a more performant way to do this, e.g. with a MappedByteBuffer or similar where we might lazily load into the ByteBuffer rather than do the work upfront?

Reply:
Updated to use mapped byte buffer.